.github/workflows/testing.yml

```
@@ -268,6 +268,76 @@ jobs:
           path: taichi-release-tests/bad-compare/*
           retention-days: 7
 
+  build_and_test_amdgpu_linux:
+    name: Build and Test (AMDGPU)
+    needs: check_files
+    timeout-minutes: ${{ github.event.schedule != '0 18 * * *' && 90 || 120 }}
+    strategy:
+      matrix:
+        tags:
+          - [self-hosted, amdgpu]
+
+    runs-on: ${{ matrix.tags }}
```

I didn't get why you are using matrix here, assuming you are just copying & pasting for convenience.
Then, please remove matrix here
```suggestion
    runs-on: [self-hosted, amdgpu]
```